### PR TITLE
doc: update FIPS provider version information

### PR DESCRIPTION
**Pull request by paulidale**
With 3.0.8 validated, we need to note this in the documentation.

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by levitte — APPROVED** [review]

**Comment by tmshort** [comment]
Shouldn't this also go into 3.0? Or is there a separate PR?

**Comment by paulidale** [comment]
3.0.8 is mentioned in our certificate.

The earlier change including version numbers didn't go into 3.0.x.  I don't remember why.  It probably should have.

**Comment by paulidale** [comment]
I've raised #21060 to include this in 3.0.x too.

**Review by t8m — APPROVED** [review]

File: doc/man7/OSSL_PROVIDER-FIPS.pod
Comment (paulidale) [reply doc/man7/OSSL_PROVIDER-FIPS.pod:460]: Including a note that 3.1 was approved was okay previously but being more specific about 3.1.1 isn't?

Comment (mattcaswell) [reply doc/man7/OSSL_PROVIDER-FIPS.pod]: I am questioning whether it was ever right

**Comment by t8m** [comment]
What about linking to https://www.openssl.org/source/ instead of mentioning concrete version info which will always be out of date.

**Comment by paulidale** [comment]
Update the text to link to the source and include an explanation.

File: doc/man7/OSSL_PROVIDER-FIPS.pod
Comment (t8m) [inline doc/man7/OSSL_PROVIDER-FIPS.pod:471]: I would just remove this.

**Comment by paulidale** [comment]
Ping @openssl/committers for second review please.

**Review by t8m — APPROVED** [review]

File: doc/man7/OSSL_PROVIDER-FIPS.pod
Comment (DDvO) [reply doc/man7/OSSL_PROVIDER-FIPS.pod:432]: And is it true that 3.0.0 has been FIPS certified? Shouldn't it be 3.1.0?

File: doc/man7/fips_module.pod
Comment (DDvO) [inline doc/man7/fips_module.pod:483]: Would it make sense to have (part of) this information also in ´NEWS.md`?

File: doc/man7/OSSL_PROVIDER-FIPS.pod
Comment (paulidale) [reply doc/man7/OSSL_PROVIDER-FIPS.pod:432]: The 3.0.8 certification replaced the 3.0.0 one because it's the same certificate.
We're working on getting 3.0.0 put back on the certificate.

It's all part of the process.

**Review by DDvO — APPROVED** [review]
LGTM

File: doc/man7/fips_module.pod
Comment (t8m) [reply doc/man7/fips_module.pod:483]: IMO this does not belong to NEWS.md

Comment (paulidale) [reply doc/man7/fips_module.pod:483]: A note saying 3.0.8 achieve FIPS validation might be worth including in NEWS.md, however it's always going to lag the release (i.e. it would be in 3.0.10 changes now).

**Comment by paulidale** [comment]
Merged, thanks for the feedback.

**Pull request closed without merging**
